tests/test_services/test_workflow_service.py

```
@@ -146,3 +151,149 @@ def test_vertex_missing_location_yields_error(self):
+        from src.backend.services.workflow_service import _acquire_workflow_slot, _release_workflow_slot, get_active_workflow_count
+        with patch("src.backend.services.workflow_service.settings") as mock_settings:
+            mock_settings.MAX_CONCURRENT_WORKFLOWS = 50
+            errors = []
+
+            def worker():
+                try:
+                    for _ in range(100):
+                        acquired = _acquire_workflow_slot()
+                        if acquired:
+                            _release_workflow_slot()
+                except Exception as e:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)